PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/common/ErrorMessageStringField.java

```diff
@@ -0,0 +1,83 @@
+package com.microsoft.azure.sparkserverless.common;
+
+import com.intellij.ui.DocumentAdapter;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.swing.*;
+import javax.swing.event.DocumentEvent;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ErrorMessageStringField extends JTextField implements isLegal {
+    @Nullable
+    private String errorMessage;
+    @Nullable
+    private Set<String> texts;
+    @NotNull
+    private final ErrorMessageTooltip errorMessageTooltip = new ErrorMessageTooltip(this);
+    @NotNull
+    private static final Pattern pattern = Pattern.compile("^[a-zA-Z0-9_-]+");
+
+    public ErrorMessageStringField() {
+        super();
+        this.getDocument().addDocumentListener(new DocumentAdapter() {
+            @Override
+            protected void textChanged(DocumentEvent e) {
```

Comment:
Can add a base class `ErrorHintedField` with these logical for both `IntergerWithErrorHintedField` and `TextWithErrorHintedField`